api/spec/requests/ingests_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+RSpec.describe "Projects API", type: :request do
+  include_context("authenticated request")
+  include_context("param helpers")
+
+  let!(:project) { FactoryBot.create(:project, draft: false) }
+  let(:project_id) { project.id }
+
+  let!(:attributes) {
+    {
+      source: markdown_source_params,
+      ingestionType: "epub"
+    }
+  }
+  let!(:valid_params) {
+    build_json_payload(attributes: attributes)
+  }
+
+  context "when a text ingest is triggered" do
+    let!(:path) { "/api/v1/projects/#{project_id}/ingest" }
+
+    before(:each) do
+      post path, headers: admin_headers, params: valid_params
+    end
+
+    it "creates and processes an ingestion" do
+      expect(response).to have_http_status(201)
+    end
+  end
+
+  let!(:text) { FactoryBot.create(:text) }
```

Comment:
This should go in the below context. Otherwise you're creating a text for the above test as well, which doesn't need it.

Likewise, the `let!(:project) { ... }` bit should go in its associated context. `valid_params` and `attributes` are shared between the two types, but text and project are specific to each nested example.

Reply:
Though also, if things are broken out into their own bespoke controllers, you should probably wind up with two very similar specs:

* spec/requests/api/v1/projects/ingestions_spec.rb
* spec/requests/api/v1/texts/ingestions_spec.rb